### PR TITLE
Mm 39663

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Add current product name to invoice so that we don't have to guess infer it in webapp system console billing history page. Make invoice item quantities a float to support metered billing.

Supported by: https://github.com/mattermost/customer-web-server/pull/661
Supports: https://github.com/mattermost/mattermost-webapp/pull/9652/files

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-39663

#### Release Note

```release-note
Add current product name to invoice
Support metered billing invoices
```

### CODE REVIEW

File: model/cloud.go [hunk 1]
```diff
@@ -138,24 +138,25 @@ func (s *Subscription) GetWorkSpaceNameFromDNS() string {
...
+	CreateAt           int64              `json:"create_at"`
+	Total              int64              `json:"total"`
+	Tax                int64              `json:"tax"`
+	Status             string             `json:"status"`
+	Description        string             `json:"description"`
+	PeriodStart        int64              `json:"period_start"`
+	PeriodEnd          int64              `json:"period_end"`
+	SubscriptionID     string             `json:"subscription_id"`
+	Items              []*InvoiceLineItem `json:"line_items"`
+	CurrentProductName string             `json:"current_product_name"`
```
Comment (neallred) [inline model/cloud.go:152]: Only change to Invoice is to add `CurrentProductName`. The rest is whitespace.

**Review by nickmisasi — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
LGTM. Marking as `Do not Merge` so we can coordinate which cloud release this will need to go into with the corresponding CWS changes

**Pull request merged**